[ADP-3407] Bump to ghc 9.8.2

paolino opened the pull request:
- [x] Bump int-cast
- [x] Bump openapi3
- [x] Remove all constraints in cabal files
- [x] Bump compiler in haskell.nix to 9.8.2

paolino committed:
Bumps for ghc 9.8.2

paolino committed:
Tweaks to allow 9.8.2 compilation

jonathanknowles commented on cabal.project:
```diff
@@ -157,6 +157,13 @@ source-repository-package
+    type: git
+    location: https://github.com/paolino/int-cast.git
+    tag: 0232b037552a4247b4a7ac21f816cc77cc95f7a4
+    --sha256: 1pmghd9pz35ycxp0c3f5sz4fghfi6ywjmp6wm9n45f8l5ah9vhhw
```
I've just made a PR to upgrade `int-cast` to support GHC versions 9.8 and 9.10 here:
- https://github.com/haskell-hvr/int-cast/pull/9

jonathanknowles commented on cabal.project:
```diff
@@ -177,8 +184,7 @@ allow-newer:
 constraints:
-    base == 4.18.2.0
-  , bimap >= 0.4.0
+  bimap >= 0.4.0
```
```suggestion
  , bimap >= 0.4.0
```

jonathanknowles reviewed (approved):
LGTM!

I've tested it locally, and:
- everything seems to build just fine. 🎉
- HLS continues to work nicely. 🎉 

Hopefully it will also work for others too!

Anviking reviewed (approved):
`cabal build all` in nix shell and HLS works on my mac

HeinrichApfelmus reviewed (approved):
> cabal build all in nix shell and HLS works on my mac

Same for me. 👍

paweljakubas reviewed (approved):
:+1:

jonathanknowles committed:
Use patched version of package `wai`.

Fixes:
https://buildkite.com/cardano-foundation/cardano-wallet/builds/5240#01900a67-445e-447e-b1bd-976ebaf665c1/39-41

Error:
```
> Network/Wai/Application/Static.hs:260:27: error: [GHC-87897]
>     • Exception when trying to run compile-time code:
>         Z:\build\wai-app-static-3.1.9\images/folder.png: withBinaryFile: invalid argument (Bad file descriptor)
>       Code: (makeRelativeToProject "images/folder.png" >>= embedFile)
>     • In the untyped splice:
>         $(makeRelativeToProject "images/folder.png" >>= embedFile)
>     |
> 260 |             L.fromChunks [$(makeRelativeToProject "images/folder.png" >>= embedFile)]
>     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

jonathanknowles committed:
Use patched version of package `fgl`.

Fixes:
```
Data/Graph/Inductive/Monad.hs:23:1: error: [GHC-87897]
    • Exception when trying to run compile-time code:
        External interpreter terminated (-15)
    • In the annotation:
        {-# ANN module "HLint: ignore Redundant lambda" #-}
   |
23 | {-# ANN module "HLint: ignore Redundant lambda" #-}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[10 of 29] Compiling Data.Graph.Inductive.NodeMap ( Data/Graph/Inductive/NodeMap.hs, dist/build/Data/Graph/Inductive/NodeMap.o )
[11 of 29] Compiling Data.Graph.Inductive.PatriciaTree ( Data/Graph/Inductive/PatriciaTree.hs, dist/build/Data/Graph/Inductive/PatriciaTree.o )
```

paolino committed:
Bump hackage dependency

paolino committed:
Bump haskell nix dependency

paolino committed:
Bump compiler to ghc 9.6.6

paolino committed:
Bump hackage index

paolino committed:
Fix new hlint issues

HeinrichApfelmus committed:
Fix compiler error in `Cardano.Read.Ledger.Block.Txs`

paolino committed:
Bump mdbook-admonish assets to 3.0.0

paolino committed:
Remove benchmarks from windows test bundle
